frontend/src/pages/SearchPage.tsx

```diff
@@ -1,8 +1,21 @@
+import { useSearchRestaurants } from '@/api/RestaurantApi';
 import { useParams } from 'react-router-dom';
 
 const SearchPage = () => {
   const { city } = useParams();
-  return <span>Search Page {city}</span>;
+  const { results } = useSearchRestaurants(city);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/37#issuecomment-2405201956>#67cbd7</a><span> - 10/10/2024, 02:13 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The useSearchRestaurants hook is called without a dependency array, which may cause unnecessary re-renders or API calls on every render.
</div>
<div id="fix">
<b>Fix:</b> Add a dependency array to the useSearchRestaurants hook call, including the 'city' variable to ensure the hook only runs when the city changes.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -6 +6 @@
 -  const { results } = useSearchRestaurants(city);
 +  const { results } = useSearchRestaurants(city, [city]);
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged